calico/openstack/mech_calico.py

```diff
@@ -600,13 +614,20 @@ def _update_port(self, context, port):
         if not port_disabled:
             LOG.info("Port enabled, attempting to update.")
 
-            with context._plugin_context.session.begin(subtransactions=True):
-                port = self.db.get_port(context._plugin_context, port['id'])
+            # When called directly from the core plugin, we don't have the ML2
+            # plugin context.
+            if getattr(context, '_plugin_context', None):
```

Comment:
@nbartos @neiljerram I'm not sure I understand how `context._plugin_context` differs from just `context` well enough to say this is OK.  Can either of you shed some light?